### PR TITLE
fix sign-compare warning b/t size_t and int

### LINKED ISSUE #294: -Wsign-compare warnings
## Description

A couple `-Wsign-compare` warnings thrown when comparing a `size_t` variable and an `int` variable.

## Steps to reproduce

1.  git clone https://github.com/borglab/gtsam
2.  mkdir build && cd build && cmake .. && make

```
[ 97%] Building CXX object gtsam_unstable/examples/CMakeFiles/SmartRangeExample_plaza1.dir/S    martRangeExample_plaza1.cpp.o
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp: In fu    nction ‘std::vector<gtsam::Event> createTrajectory(int)’:
 /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp:60:28:     warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long uns    igned int’} and ‘int’ [-Wsign-compare]
      60 |   for (size_t key = 0; key < n; key++) {                                  
         |                        ~~~~^~~                                            
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp: In fu    nction ‘gtsam::Values createInitialEstimate(int)’:
   /home/acxz/vcs/git/github/acxz/gtsam/gtsam_unstable/examples/TimeOfArrivalExample.cpp:117:28    : warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long un    signed int’} and ‘int’ [-Wsign-compare]
     117 |   for (size_t key = 0; key < n; key++) {                                  
         |                        ~~~~^~~          
```

## Expected behavior

No `-Wsign-compare` warnings.

## Environment

OS: ArchLinux
GCC: 9.3.0
GTSAM: latest head on develop branch at time of posting: 
b4eb0c2

## Additional information

N/A

### PR DESCRIPTION
**Pull request by acxz**
Resolves #294

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/295)

### CODE REVIEW

**Comment by jlblancoc** [comment]
Why not using C++ `static_cast<size_t>()` instead of old `C` type cast? With some static analyzers, the latter would even trigger new warnings :-)

**Comment by acxz** [comment]
Ah yes, static_casting is safer. Thank you!

File: gtsam_unstable/examples/TimeOfArrivalExample.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ vector<Event> createTrajectory(int n) {
   vector<Event> trajectory;
   double timeOfEvent = 10;
   // simulate emitting a sound every second while moving on straight line
-  for (size_t key = 0; key < n; key++) {
+  for (size_t key = 0; key < static_cast<size_t>(n); key++) {
```
Comment (dellaert) [inline gtsam_unstable/examples/TimeOfArrivalExample.cpp:60]: Why not simply change the argument type to size_t ?

Comment (acxz) [reply gtsam_unstable/examples/TimeOfArrivalExample.cpp:60]: Ahh yeah looking at the inputs for when these methods are called, `n` is of type `size_t`.

Comment (acxz) [reply gtsam_unstable/examples/TimeOfArrivalExample.cpp]: Pushed the fix out.

**Comment by varunagrawal** [comment]
Straightforward. LGTM.

**Pull request merged**